src/fmu/dataio/_model/specification.py

```diff
@@ -96,6 +96,21 @@ class CPGridSpecification(RowColumnLayer):
     zscale: float = Field(allow_inf_nan=False)
     """Scaling factor for the z-axis."""
 
+    zonation: Optional[List[ZoneDefinition]] = Field(default=None)
+    """
+    Zone names and corresponding layer index ranges. The list is ordered from
+    shallowest to deepest zone. Note the layer indices are zero-based; add 1 to
+    convert to corresponding layer number.
+    """
+
+
+class ZoneDefinition(BaseModel):
+    """Zone name and corresponding layer index min/max"""
+
+    name: str
+    min_layer_idx: int = Field(ge=0)
+    max_layer_idx: int = Field(ge=0)
```

Comment:
Can you docstring beneath these fields? They will be pulled into the documentation